handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -1294,6 +1266,11 @@ private HandshakeStatus handshake() throws SSLException {
         // Adding the OpenSslEngine to the OpenSslEngineMap so it can be used in the AbstractCertificateVerifier.
         engineMap.add(this);
 
+        if (!certificateSet && keyMaterialManager != null) {
+            certificateSet = true;
+            keyMaterialManager.setServerKeyMaterial(this);
```

Comment:
this feels a bit awkward to call a method where the name implies it only works on the client or server in the SslEngine. Do you agree and would getting rid of the `Server`/`Client` in the `set[Client|Server]KeyMaterial` methods be an improvement?